src-opam/dockerfile_opam.ml

```diff
@@ -81,23 +81,32 @@ let bubblewrap_and_dev_packages distro =
     @@ dev_packages ()
 
 let install_bubblewrap_wrappers =
-  (* Enable bubblewrap *)
-  run "echo 'wrap-build-commands: []' > ~/.opamrc-nosandbox" @@
-  run "echo 'wrap-install-commands: []' >> ~/.opamrc-nosandbox" @@
-  run "echo 'wrap-remove-commands: []' >> ~/.opamrc-nosandbox" @@
-  run "echo 'required-tools: []' >> ~/.opamrc-nosandbox" @@
-  run "echo '#!/bin/sh' > /home/opam/opam-sandbox-disable" @@
-  run "echo 'cp ~/.opamrc-nosandbox ~/.opamrc' >> /home/opam/opam-sandbox-disable" @@
-  run "echo 'echo --- opam sandboxing disabled' >> /home/opam/opam-sandbox-disable" @@
+  let strip = true in
+  let opamrc_sandbox = heredoc ~strip
+{|	wrap-build-commands: ["%%{hooks}%%/sandbox.sh" "build"]
+	wrap-install-commands: ["%%{hooks}%%/sandbox.sh" "install"]
+	wrap-remove-commands: ["%%{hooks}%%/sandbox.sh" "remove"]|} in
+  let opamrc_nosandbox = heredoc ~strip
+{|	wrap-build-commands: []
+	wrap-install-commands: []
+	wrap-remove-commands: []
+	required-tools: []|} in
+  let sandbox_enable = heredoc ~strip
+{|	#!/bin/sh
+	cp ~/.opamrc-sandbox ~/.opamrc
+	echo --- opam sandboxing enabled|} in
+  let sandbox_disable = heredoc ~strip
+{|	#!/bin/sh
+	cp ~/.opamrc-nosandbox ~/.opamrc
+	echo --- opam sandboxing disabled|} in
+  (* Disable bubblewrap *)
+  copy_heredoc ~chown:"opam" ~src:[opamrc_nosandbox] ~dst:"/home/opam/.opamrc-nosandbox" () @@
+  copy_heredoc ~chown:"opam" ~src:[sandbox_disable] ~dst:"/home/opam/opam-sandbox-disable" () @@
```

Comment:
IMHO, this part of the syntax is harder to read than the original version.  The `{| |}` introduces complexity by mixing ocaml and bash syntax instead of the `run "cmd"`.

Reply:
Thank you for the review Etienne. I'd say the complexity is having to follow the file line-by-line through multiple `RUN` and `echo >` calls and that the code is more "readable", or at least that the file isn't being split and stands out in the OCaml as well as in the generated Dockerfile.
Since you don't share that opinion, but also that I think we can agree it's a minor change without much technical advantages, I'm going to split the commit using heredocs (even though it was the feature that motivated this change) from this PR so that maybe it can move forward.